src/cryptoadvance/specter/templates/includes/recipient-box.html

```diff
@@ -109,52 +124,48 @@
 			this.removeElement = clone.getElementById('remove'); 
 
             this.shadowRoot.appendChild(clone);
-			this.addAssetSelectorToShadowRoot('', 'btc')			
+			this.addAssetSelectorToShadowRoot()			
 		}
 
 
 
 		// add the html code for the assetSelector
-		addAssetSelectorToShadowRoot(amount, amountUnit) { 
-			if (amount >= 0){
-				this.amountElement.value = amount;
-			}
-			
-
-			if (amountUnit == 'sat') {
-				amount = parseFloat(Number.parseFloat(amount * 1e8).toFixed(0));
-				if (amount == 'NaN') {
-					amount = '-'
-				}
-			}
-			let step = 1;
-			if (amountUnit == 'btc') {
-				step = 1e-8;
-			}
-			let satChecked = amountUnit == 'btc' ? '' : 'checked';
-			let btcChecked = amountUnit == 'btc' ? 'checked' : '';
-			let convertedUnit = amountUnit == 'btc' ? 'sat' : 'BTC';
-			this.shadowRoot.getElementById("converted_unit_label").innerText = convertedUnit;
-			
+		addAssetSelectorToShadowRoot() { 
+			// add different asset selector template for bitcoin and liquid
 			{% if specter.is_liquid and wallet.balance.get("assets", {}) %}
-				let assetSelector = `
-					<select name="amount_unit" style="width: 100px;" >
-						<option value="btc">LBTC</option>
-						<option value="sat">L-sat (10⁻⁸ LBTC)</option>
-					{% for asset in wallet.balance.get("assets",{}).keys() | sort %}
-						<option value="{{asset}}">{{asset | assetlabel}}</option>
-					{% endfor %}
-					</select>
-					`;				
+				var template_content = document.getElementById('liquid-asset-selector').content;
+				var clone = template_content.cloneNode(true);				
+
+				this.amountUnitElement = clone.getElementById('amount_unit');
+				// create custom getter and setter functions
+				this.setUnit =  (newValue) => {
+					this.amountUnitElement.value = newValue;
+					this.updateUnitLabelAndStep();
+				}
+				this.getUnit =  () => {
+					return this.amountUnitElement.value;
+				}
 			{% else %}
-				let assetSelector = `
-					<label><input type="radio" class="inline" style="margin: 0 5px;" name="amount_unit"  value="sat" ${satChecked}>sat</label>
-					<label><input type="radio" class="inline" style="margin: 0 5px;" name="amount_unit"  value="btc" ${btcChecked}>BTC</label>
-					`;
+				var template_content = document.getElementById('bitcoin-asset-selector').content;
+				var clone = template_content.cloneNode(true);				
+
+				this.amountUnitSatElement = clone.getElementById('amount_unit_sat');
+				this.amountUnitBtcElement = clone.getElementById('amount_unit_btc');
+				// create custom getter and setter functions
+				this.setUnit =  (newValue) => {
+					this.amountUnitBtcElement.checked =  newValue == 'btc';
+					this.amountUnitSatElement.checked =  newValue == 'sat';
+					this.updateUnitLabelAndStep();
+				}
+				this.getUnit =  () => {
+					return this.amountUnitBtcElement.checked  ? this.amountUnitBtcElement.value : this.amountUnitSatElement.value ;
+				}
+
+				// set the initial unit at creation
```

Comment:
I think this comment should rather be below at in the same file:

```
		// unit
		set unit(newValue){
			// this function is different for bitcoin and liquid 
			this.setUnit(newValue);
		}			
		get unit(){
			// this function is different for bitcoin and liquid 
			return this.getUnit();
		}	
```

This is what is called from "the outside", e.g. in `wallet_send` now with:
`newRecipient.unit = amount_unit;`

Reply:
I am not sure I understand.  Do you mean the comment `// set the initial unit at creation` should be in `recipient_box.html`?

Reply:
Or put it in `wallet_send` where `newRecipient.unit = amount_unit`; it just looks a bit lost here. Can IMO also just be deleted.

Reply:
Oh I missed that, yes in https://github.com/cryptoadvance/specter-desktop/pull/1782/commits/b3ff4d88ed0bbff8ce1585b94700ab191c35e6f2